### PR TITLE
Preliminary work for a future change in outputting results that uses less memory

### PR DESCRIPTION
**Pull request by meeuw**
## Description
Preliminary work for a future change in outputting results that uses less memory

## Checklist

- [x] I've added this contribution to the `changelog.md`.

### CODE REVIEW

**Comment by j-bennet** [comment]
@meeuw Is this WIP or done?

**Comment by meeuw** [comment]
done, please review and merge.

File: changelog.rst [hunk 1]
```diff
@@ -9,6 +9,7 @@ Features:
 * Don't quote identifiers that are non-reserved keywords. (Thanks: `Joakim Koljonen`_)
 * Remove the ``...`` in the continuation prompt and use empty space instead. (Thanks: `Amjith Ramanujam`_)
 * Add \conninfo and handle more parameters with \c (issue #716) (Thanks: `François Pietka`_)
+* Preliminary work for a future change in outputting results that uses less memory. (Thanks: `Dick Marinus`_)
```
Comment (j-bennet) [inline changelog.rst]: This should be in "Internal changes" section.

File: pgcli/main.py [hunk 2]
```diff
@@ -1011,16 +1012,26 @@ def format_arrays(data, headers, **_):
         headers = [case_function(utf8tounicode(x)) for x in headers]
         rows = list(cur)
         formatted = formatter.format_output(rows, headers, **output_kwargs)
-        first_line = formatted[:formatted.find('\n')]
+
+        if isinstance(formatted, (text_type)):
```
Comment (j-bennet) [inline pgcli/main.py]: Can this be anything other than `text_type`? If yes, what happens in that case?

Comment (meeuw) [inline pgcli/main.py]: This is exactly what this PR is about (you have a sharp eye!). In the next version of cli_helpers a generator will be outputted from format_output which yields lines without newline characters.

When this new version of cli_helpers is released these lines can be removed. See https://github.com/dbcli/cli_helpers/pull/13

**Comment by j-bennet** [comment]
👍

**Pull request merged**